### PR TITLE
DEPS: Bump optional dependencies

### PR DESCRIPTION
**Pull request by mroeschke**
These versions will have been around for ~1 year once 2.2 comes out

### CODE REVIEW

File: pandas/tests/io/pytables/test_file_handling.py [hunk 1]
```diff
@@ -236,8 +236,12 @@ def test_complibs_default_settings_override(tmp_path, setup_path):
     # with xfail, would sometimes raise UnicodeDecodeError
     # invalid state byte
 )
-def test_complibs(tmp_path, lvl, lib):
+def test_complibs(tmp_path, lvl, lib, request):
     # GH14478
+    if not PY311 and is_platform_linux() and lib == "blosc2":
```
Comment (mroeschke) [inline pandas/tests/io/pytables/test_file_handling.py:241]:
```suggestion
    if PY311 and is_platform_linux() and lib == "blosc2":
```

File: pandas/tests/io/pytables/test_file_handling.py [hunk 2]
```diff
@@ -236,8 +236,12 @@ def test_complibs_default_settings_override(tmp_path, setup_path):
     # with xfail, would sometimes raise UnicodeDecodeError
     # invalid state byte
 )
-def test_complibs(tmp_path, lvl, lib):
+def test_complibs(tmp_path, lvl, lib, request):
     # GH14478
+    if PY311 and is_platform_linux() and lib == "blosc2":
```
Comment (mroeschke) [inline pandas/tests/io/pytables/test_file_handling.py:241]:
```suggestion
    if PY311 and is_platform_linux() and lib == "blosc2" and lvl != 0:
```

File: pandas/tests/interchange/test_impl.py [hunk 3]
```diff
@@ -309,11 +313,21 @@ def test_datetimetzdtype(tz, unit):
     tm.assert_frame_equal(df, from_dataframe(df.__dataframe__()))
 
 
-def test_interchange_from_non_pandas_tz_aware():
+def test_interchange_from_non_pandas_tz_aware(request):
     # GH 54239, 54287
     pa = pytest.importorskip("pyarrow", "11.0.0")
     import pyarrow.compute as pc
 
+    if is_platform_windows() and is_ci_environment():
```
Comment (jbrockmendel) [inline pandas/tests/interchange/test_impl.py:321]: use _require_timezone_database?

Comment (mroeschke) [reply pandas/tests/interchange/test_impl.py:321]: Yeah can do this in a follow up. It's current centralized in `test_arrow.py`

File: ci/deps/actions-310.yaml [hunk 4]
```diff
@@ -24,39 +24,41 @@ dependencies:
...
-  - qtpy>=2.2.0
-  - openpyxl>=3.0.10
-  - pandas-gbq>=0.17.5
-  - psycopg2>=2.9.3
-  - pyarrow>=7.0.0
+  - qtpy>=2.3.0
+  - openpyxl>=3.1.0
+  # Doesn't install well with pyarrow
+  # https://github.com/pandas-dev/pandas/issues/55525
+  # - pandas-gbq>=0.19.0
```
Comment (WillAyd) [inline ci/deps/actions-310.yaml:45]: Does this not affect test coverage?

Comment (mroeschke) [reply ci/deps/actions-310.yaml:45]: Not really. There's only one unit test that just attempts to import this package which doesn't seem useful

**Comment by WillAyd** [comment]
I think this looks good. Is the 1 year dependency minimum new? That was 2 in the past right?

**Comment by mroeschke** [comment]
> I think this looks good. Is the 1 year dependency minimum new? That was 2 in the past right?

I bumped our optional dependencies to a 1  year min last time https://github.com/pandas-dev/pandas/pull/52552. Our required ones usually follow 2

**Review by WillAyd — APPROVED** [review]

File: pandas/compat/_optional.py [hunk 5]
```diff
@@ -15,42 +15,42 @@
...
-    "openpyxl": "3.0.10",
-    "pandas_gbq": "0.17.5",
-    "psycopg2": "2.9.3",  # (dt dec pq3 ext lo64)
+    "openpyxl": "3.1.0",
+    "pandas_gbq": "0.19.0",
+    "psycopg2": "2.9.6",  # (dt dec pq3 ext lo64)
     "pymysql": "1.0.2",
-    "pyarrow": "7.0.0",
-    "pyreadstat": "1.1.5",
+    "pyarrow": "10.0.1",
```
Comment (jorisvandenbossche) [inline pandas/compat/_optional.py:37]: FWIW, the pyarrow PDEP said "When the minimum version of PyArrow is bumped, PyArrow will be bumped to the highest version that has been released for at least 2 years.". But pyarrow 10.0 was only released 1 year ago.

(and in general I also think 1 year is too short for pyarrow)

**Pull request merged**